module_auto_update/models/module.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    checksum_dir = fields.Char(
+        compute='_compute_checksum_dir',
+    )
+    checksum_installed = fields.Char()
+
+    @api.depends('name')
+    def _compute_checksum_dir(self):
+        exclude = self.env["ir.config_parameter"].get_param(
+            "module_auto_update.checksum_excluded_extensions",
+            "pyc,pyo"
+        ).split(",")
+
+        for r in self:
+            r.checksum_dir = dirhash(
+                get_module_path(r.name),
+                'sha1',
+                excluded_extensions=exclude
+            )
+
+    def _store_checksum_installed(self, vals):
+        if ('checksum_installed' not in vals and
```

Comment:
Simplify this conjoined if by moving `not self.env.context` above, inverse the logic, and return.